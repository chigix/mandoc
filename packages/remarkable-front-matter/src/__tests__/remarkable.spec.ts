import { readFixture } from '@fixtures';
import * as Remarkable from 'remarkable';
import frontMatter from '..';

const remarkable = new Remarkable();
remarkable.use(frontMatter);

test('front-matter-plugin', () => {
  expect(remarkable.render(readFixture('ipsum.md')))
    .toMatchSnapshot();
  const env = { frontMatter: undefined };
  expect(remarkable.render(readFixture('markdown-cases/front-matter-case.md'), env))
    .toMatchSnapshot();
  expect(env.frontMatter).toMatchSnapshot();
});
