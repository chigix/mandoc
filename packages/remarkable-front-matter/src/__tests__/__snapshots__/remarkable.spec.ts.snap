// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`front-matter-plugin 1`] = `
"<h1>A First Level Header</h1>
<h2>A Second Level Header</h2>
<h3>A Third Level Header</h3>
<h4>A Fourth Level Header</h4>
<h5>A Fifth Level Header</h5>
<h6>A Sixth Level Header</h6>
<p>This is a paragraph. Cupcake ipsum dolor sit amet. Tiramisu liquorice gummies powder. Biscuit pastry pastry cake candy gummies carrot cake gingerbread. Gummi bears tart cupcake bonbon sugar plum biscuit. Icing marzipan tootsie roll. Pie gingerbread dragée wypas cookie. Faworki bonbon topping sweet soufflé jelly.</p>
<p>And another  paragraph. Jelly beans tootsie roll fruitcake toffee wypas chupa chups pudding. Soufflé muffin halvah powder sugar plum cake. Toffee cookie biscuit faworki pastry. Bonbon gummi bears topping caramels marshmallow. Chupa chups candy canes toffee. Powder powder candy canes pastry. Tootsie roll marshmallow powder.</p>
<p><em>These words are emphasized</em>
<em>Also these words are emphasized</em></p>
<p><strong>These words are strong emphasized</strong>
<strong>Also these words are strong emphasized</strong></p>
<blockquote>
<p>This is a very simple blockquote</p>
</blockquote>
<ul>
<li>This</li>
<li>is</li>
<li>an</li>
<li>unordered</li>
<li>list.</li>
</ul>
<p>And...</p>
<ol>
<li>This</li>
<li>is</li>
<li>an</li>
<li>ordered</li>
<li>list.</li>
</ol>
<p><a href=\\"http://example.com/\\">This is an example link</a>.
<a href=\\"http://example.com/\\" title=\\"¡Buenos días!\\">This is an example link with a title</a>.</p>
<p>This is a image
<img src=\\"http://placehold.it/350x150\\" alt=\\"Alt\\" title=\\"I'm an image\\"></p>
<p><code>this is a line of code</code></p>
<pre><code>And these are several lines of code. Pudding sesame snaps cupcake bonbon cupcake. Icing macaroon donut.

More code. Donut cheesecake chocolate cake. Tart pudding wafer dessert chocolate bar jelly-o apple pie dessert. Carrot cake apple pie pie liquorice oat cake lollipop pastry caramels. Jelly-o sesame snaps gummi bears.

And more. Sugar plum lemon drops chupa chups chocolate pastry. Faworki applicake applicake brownie topping liquorice chocolate liquorice icing. Cake pudding pudding cake candy sugar plum soufflé.\`
</code></pre>
<pre><code class=\\"language-js\\">var Remarkable = require('remarkable');
var hljs       = require('highlight.js') // https://highlightjs.org/

// Actual default values
var md = new Remarkable({
  highlight: function (str, lang) {
    if (lang &amp;&amp; hljs.getLanguage(lang)) {
      try {
        return hljs.highlight(lang, str).value;
      } catch (err) {}
    }

    try {
      return hljs.highlightAuto(str).value;
    } catch (err) {}

    return ''; // use external default escaping
  }
});
</code></pre>
<p>And that's all ...
This is an <strong>extremely simple and basic</strong> guide. To understand a lot better this kind of dark magic called markdown, logically I recommend the official wiki of the creator. Go, run, jump to <a href=\\"http://daringfireball.net/projects/markdown/\\">the site of John Gruber</a>.</p>
"
`;

exports[`front-matter-plugin 2`] = `
"<h1>Meeting Report August</h1>
<h2>Steady-state Stochastic Processes</h2>
<p>CONTENTS...</p>
"
`;

exports[`front-matter-plugin 3`] = `
Object {
  "author": Array [
    Object {
      "Richard Lea": Object {
        "url": "https://github.com/chigix",
      },
    },
  ],
  "date": 2018-07-04T15:15:19.000Z,
  "tags": Array [
    "meeting",
    "report",
    "kanban",
    "lean",
  ],
  "title": "Meeting Report 2018-8-3",
}
`;
